File: python_ruby workspace/OOP_variable4.py
class C:
    def __init__(self,v) :
        self.__value=v 
        #언더바 두개 앞에 붙히면 인스턴스 밖에서는 접근 못하게 막음
    def show(self) :
        print(self.__value)
                
c1 = C(10)
c1.show()
# print(c1.__value)
# c1.__value=24
# print(c1.__value)

File: python_ruby workspace/OOP_variable4.rb
class C
    attr_reader :value #외부에서 value 접근가능하게(읽게)
    attr_writer :value #외부에서 value 접근가능하게(쓰게)
    attr_accessor :value #외부에서 value 접근가능하게(읽고 쓰게)
    def initialize(v)
        @value=v
    end
    def show()
        p @value
    end
end
                
c1 = C.new(10)
p c1.value
c1.value=20
p c1.value